### ISSUE TITLE
unfinished statement `Future<List<>>` breaks the IDE 

### ISSUE THREAD

**Issue by gsouf** [flutter/flutter-intellij#4862]
## Steps to Reproduce

- Open android studio
- Open an existing project
- Create a new dart file
- add the following statement `Future<List<>>` (yes it's unfinished on the purpose, that's how it breaks android studio)
- go to another dart file, try autocompletion (example type `fin` and hit `ctrl`+`space` in order to autocomplete the word `final`)

The whole IDE completion is now broken - not only that, but also slower and laggy - code analysis is also broken.

- sometimes it does not break straight away so try to restart the ide as it is (leave in place the unfinished statement in the new file).
- when android studio restarts, then autocompletion does not work

Now to fix this you need to remove the unfinished statement and to restart the ide again.

Note that I have tried with a few other statement combinations, that also broke the ide. ie:

```
List<Future<>>
List<List<>>
List<List<List<>>>
```

Whereas the following do not break:

```
List<>
```

## Version info

```
[√] Flutter (Channel stable, 1.20.4, on Microsoft Windows [version 10.0.18362.1082], locale fr-FR)
    • Flutter version 1.20.4 at C:\Users\bob\Documents\flutter
    • Framework revision fba99f6cf9 (5 days ago), 2020-09-14 15:32:52 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2


[√] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at C:\Users\bob\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
```

**Comment by pq** [comment]
Thanks for the report @gsouf!

I'm seeing this locally too.

@scheglov: I'm not seeing anything in a quick scan through the logs...  Curious if you have any thoughts?

**Comment by scheglov** [comment]
It seems to the into the infinite loop in `IgnoreInfo.forDart`.

![image](https://user-images.githubusercontent.com/384794/94312127-3a035a00-ff31-11ea-9947-eb5b4bdd9f23.png)

**Comment by bwilkerson** [comment]
This is not specific to either Flutter or IntelliJ, so I'm going to try to move the issue.

I was able to reproduce the issue from the command-line, but have not yet been able to cause it to fail in the IDE so that I can debug it to figure out what's causing the issue. I did notice some strange behavior from the parser, for which I opened a separate issue (https://github.com/dart-lang/sdk/issues/43570).
